libraries/AP_PiccoloCAN/AP_PiccoloCAN.h

```diff
@@ -0,0 +1,126 @@
+/*
+ * This file is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This file is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Author: Oliver Walters
+ */
+
+#ifndef AP_PICCOLOCAN_H_
+#define AP_PICCOLOCAN_H_
+
+#include <AP_HAL/AP_HAL.h>
+#include <AP_HAL/CAN.h>
+#include <AP_HAL/Semaphores.h>
+
+#include "piccolo_protocol/ESCPackets.h"
+
+// maximum number of ESC allowed on CAN bus simultaneously
+#define PICCOLO_CAN_MAX_NUM_ESC 8
```

Comment:
We can support 64 simultaneous ESCs with the 4-in-1 command packets. I have this defined as 8 currently because I could only see 8 motor channels available in the code? 

![image](https://user-images.githubusercontent.com/10080325/70282882-6966a000-1813-11ea-94fe-63933ae86b4a.png)

Although I now see some more channels here:

![image](https://user-images.githubusercontent.com/10080325/70283026-df6b0700-1813-11ea-9206-7a9d6f2d3fb5.png)

Is 12x motor channels the maximum supported?

Reply:
I have (for now) increased the number of supported ESC channels to 12

Reply:
> We can support 64 simultaneous ESCs with the 4-in-1 command packets. I have this defined as 8 currently because I could only see 8 motor channels available in the code?

12 is max for VTOL motors, but uses may have some fwd motors too. Maximum possible config at the moment would be 15 motors with 2 fwd motors plus a boost motor plus 12 vertical (though nobody has that config).
We'd need to use a different way to map motors to do that though. Your current scheme has the big advantage of simplicity. If you look at what I did for DShot and BLHeli32 you can see a more complex scheme that allows for more flexibility (and confusion ....)

Reply:
I should also note that the scheme you're using now doesn't allow for duplicates. There are some occasions where users want two physical ESCs for one logical output. I don't think it's an important enough case to redo this PR now, but worth thinking about for future schemes